gtda/images/filtration.py

```diff
@@ -0,0 +1,150 @@
+"""Binary image filtration."""
+# License: GNU AGPLv3
+
+import numbers
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from joblib import Parallel, delayed, effective_n_jobs
+from sklearn.utils import gen_even_slices
+from sklearn.utils.validation import check_is_fitted, check_array
+from ..utils._docs import adapt_fit_transform_docs
+from ..utils.validation import validate_params
+
+
+@adapt_fit_transform_docs
+class HeightFiltration(BaseEstimator, TransformerMixin):
+    """Transformer returning a collection of direction-based height image filtrations
+    from a collection of 2D or 3D boolean images.
+
+    Parameters
+    ----------
+    direction : ndarray, shape (n_dimensions_, [1]), optional, default
```

Comment:
Is it correct that the parameter would depend on an attribute calculated in `fit`?

Reply:
Its shape is the dimension of the input images. Let me rewrite that to clear out this ambiguity.